### PR TITLE
Test and fix `norm_equation` for `RelNumFieldOrder`

### PR DESCRIPTION
**Pull request by fingolfin**
While looking at PR #4261 I noticed the norm equation code for `RelNumFieldOrder` is untested. So I though I could just quickly add a simple test. Sadly it revealed that that code is unrunnable. I applied the first obviously necessary tweak. But it is not enough: there is no `Hecke.absolute_field` method which takes a `Hecke.RelSimpleNumField{AbsSimpleNumFieldElem}` (and the methods which do exist seem to have two return values, not three as the code here expects).

I think it is more efficient if @fieker has a look at resolving this, he may know immediately what's going on, unlike me :-).

Oh yeah, and next up missing would be a test for `is_irreducible(a::AbsSimpleNumFieldOrderElem)` but I didn't even try to go there 😂

### CODE REVIEW

File: src/NumberTheory/NmbThy.jl [hunk 1]
```diff
@@ -129,7 +129,7 @@ norm_equation(R::AbsNumFieldOrder, k::Base.Integer; abs::Bool = false) = norm_eq
 function norm_equation_fac_elem(R::Hecke.RelNumFieldOrder{AbsSimpleNumFieldElem,Hecke.AbsSimpleNumFieldOrderFractionalIdeal}, a::AbsNumFieldOrderElem{AbsSimpleNumField,AbsSimpleNumFieldElem})
 
   @assert Hecke.is_maximal(R)
-  Ka, mKa, mkK = absolute_field(Hecke.nf(R))
+  Ka, mKa, mkK = Hecke.absolute_field(Hecke.nf(R))
```
Comment (fingolfin) [inline src/NumberTheory/NmbThy.jl]: So this line needs to be changed further

Comment (fingolfin) [reply src/NumberTheory/NmbThy.jl]: I hope @fieker or @thofma can tell me what the right replacement is here (or just take care of it)

Comment (thofma) [inline src/NumberTheory/NmbThy.jl]:
```suggestion
  Ka, mKa, mkK = collapse_top_layer(Hecke.nf(R))
```

File: src/NumberTheory/NmbThy.jl [hunk 2]
```diff
@@ -144,9 +144,9 @@ function norm_equation_fac_elem(R::Hecke.RelNumFieldOrder{AbsSimpleNumFieldElem,
   q, mms = snf(q)
   mq = mq*inv(mms)
 
-  C = vcat([matrix(ZZ, 1, ngens(q), [valuation(mS(preimage(mq, q[i])), p) for i=1:ngens(q)]) for p = keys(lp)])
+  C = vcat([matrix(ZZ, 1, ngens(q), [valuation(mS(preimage(mq, q[i])), p) for i=1:ngens(q)]) for p = keys(lp)]...)
```
Comment (lgoettgens) [inline src/NumberTheory/NmbThy.jl]:
```suggestion
  C = reduce(vcat, (matrix(ZZ, 1, ngens(q), [valuation(mS(preimage(mq, q[i])), p) for i=1:ngens(q)]) for p = keys(lp)))
```
no need to allocate the vector here. and `reduce` is preferred to not compile many different dispatches of `vcat` just for different number of arguments

File: src/NumberTheory/NmbThy.jl [hunk 3]
```diff
@@ -144,9 +144,9 @@ function norm_equation_fac_elem(R::Hecke.RelNumFieldOrder{AbsSimpleNumFieldElem,
...
   
-  A = vcat([matrix(ZZ, 1, ngens(q), [valuation(norm(mkK, mS(preimage(mq, g))), p) for g in gens(q)]) for p = keys(la)])
+  A = vcat([matrix(ZZ, 1, ngens(q), [valuation(norm(mkK, mS(preimage(mq, g))), p) for g in gens(q)]) for p = keys(la)]...)
```
Comment (lgoettgens) [inline src/NumberTheory/NmbThy.jl]:
```suggestion
  A = reduce(vcat, (matrix(ZZ, 1, ngens(q), [valuation(norm(mkK, mS(preimage(mq, g))), p) for g in gens(q)]) for p = keys(la)))
```

**Comment by fingolfin** [comment]
@fieker @thofma OK to merge? Then please approve ;-)

**Review by fieker — APPROVED** [review]

**Pull request merged**